src/include/duckdb/optimizer/join_order/query_graph_manager.hpp

```diff
@@ -48,13 +48,16 @@ class FilterInfo {
 
 public:
 	unique_ptr<Expression> filter;
-	JoinRelationSet &set;
+	optional_ptr<JoinRelationSet> set;
```

Comment:
Nitpick: could this be a `reference<JoinRelationSet>` instead? This reference can be changed to refer to something else later on, which I think you're doing here. The only difference with `optional_ptr` is that it must always be initialized and, therefore, cannot be `nullptr`.